coderd/externalauth/externalauth_test.go

```diff
@@ -289,6 +290,11 @@ func TestRefreshToken(t *testing.T) {
 		require.NoError(t, err)
 		require.True(t, ok)
 		require.True(t, updated.OAuthExtra.Valid)
+		extra := map[string]interface{}{}
+		require.NoError(t, json.Unmarshal(updated.OAuthExtra.RawMessage, &extra))
+		mapping, ok := extra["authed_user"].(map[string]interface{})
+		require.True(t, ok)
+		require.Equal(t, updated.OAuthAccessToken, mapping["access_token"])
```

Comment:
This is 100% ok.

I am thinking of maybe exposing an easy way to hit the IDP with a Bearer token that would essentially do this check. The `/userinfo` will validate an access token for the request, but `oauth2` is a subset of `OIDC` and does not have an easy `UserInfo` method to call from the token/config.

Just would be a nice helper to have like `fake.UserInfo(accessToken string)` or something convenient. Maybe :thinking: